In [1]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [20]:
#Set up random set

N = 30

x1 = np.linspace(0,10,N)

X = np.column_stack((np.ones((N,1)),x1,x1**2,np.sqrt(x1)))
W_true = np.array([2.974, 8.2928, 1.23748,4.3478,3.5689,2.643,6.1065,3.5786]).reshape(4,2)
Y = X.dot(W_true) + np.random.randn(N,2) * 20

In [21]:
#Find w


w = np.linalg.solve(X.T.dot(X), X.T.dot(Y))
w

array([[ -3.61263528,  32.69839782],
       [  2.31808731,  33.03018796],
       [  3.21878505,   1.32455883],
       [ 14.15228938, -57.74426425]])

In [22]:
#Find y_hat

Y_hat = X.dot(w)
Y_hat

array([[ -3.61263528,  32.69839782],
       [  5.87994871,  10.33701635],
       [ 11.2698145 ,   8.15382876],
       [ 16.62421002,   9.55362327],
       [ 22.32947647,  12.96004794],
       [ 28.53525987,  17.76251611],
       [ 35.31830781,  23.64788177],
       [ 42.72421188,  30.43012184],
       [ 50.78267752,  37.98803197],
       [ 59.51433819,  46.23742335],
       [ 68.934217  ,  55.11699861],
       [ 79.05365518,  64.58048319],
       [ 89.88146374,  74.59192635],
       [101.42464961,  85.12273838],
       [113.68889399,  96.14973882],
       [126.67887898, 107.65382368],
       [140.39851756, 119.6190272 ],
       [154.85111963, 132.03184437],
       [170.03951461, 144.88073053],
       [185.96614379, 158.1557247 ],
       [202.633131  , 171.84816122],
       [220.04233758, 185.95044545],
       [238.19540566, 200.45587717],
       [257.09379268, 215.35850972],
       [276.73879923, 230.65303638],
       [297.13159173, 246.33469797],
       [318.27322102, 262.39920685],
 

In [23]:
#Finnally we will find our R^2


def R2(Y,Y_hat):
    return 1 - np.sum((Y-Y_hat)**2)/np.sum((Y-Y.mean())**2)
R2_array=[]
for k in range(2):
    R2_array.append(R2(Y[:,k],Y_hat[:,k]))

R2_array

[0.9756314525214654, 0.9653654466271347]

Your R^2 for Y_hat_1 [0.9756314525214654 and your R^2 for y_hat_2 is 0.9653654466271347]. This is pretty good. 
#(but I mean, I'm coming up with the data and random noise so of course it is)